core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -661,6 +687,20 @@ public boolean accept(CarbonFile file) {
         }
       });
       if (listFiles != null && listFiles.length > 0) {
+        Set<String> mergedAndInvalidIndexFiles = getInvalidAndMergedIndexFiles(
+            Arrays.stream(listFiles).map(file -> file.getAbsolutePath())
+                .collect(Collectors.toList()));
+        // Delete index files that are merged.
+        for (CarbonFile indexFile : listFiles) {
+          if (mergedAndInvalidIndexFiles.contains(indexFile.getAbsolutePath())) {
+            indexFile.delete();
```

Comment:
`getSegmentFileForPhysicalDataPartitions ` method is called for only Alter add hive partition flow and as it is not old store, it can be deleted. We are triggering `alterTableMergeIndexEvent` in this flow, the index files are not deleted immediately after merge index because `isOldStoreIndexFilesPresent` flag is set true in this event. 
I didn't want to change the flag for Alter add hive partition flow as we may actually add old store data(store <= 1.1 version) and it has to read all blocklet info from the file footer of carbondata file.